docs/source/index.rst

```diff
@@ -257,6 +257,7 @@ Deployment Guides and Tutorials
 
     introduction/index
     getting-started/index
+    getting-started/tracking-server-overview/index
```

Comment:
We'll want to remove this and instead add a new section here: https://github.com/mlflow/mlflow/blob/e90a2c8ee0160006a83930f68899d8771e381877/docs/source/getting-started/index.rst?plain=1#L117

1. H2 heading "MLflow Infrastructure Management Guides" or something similar
2. a new :hidden: toctree at the end of the new section that has the reference to `tracking-server-overview/index` 
3. A new simple-grid card entry for this section that has a single card pointing to the guide index.rst from this PR. 

We should be good to go after those changes!